src/perf_counters.h

```diff
@@ -94,15 +94,15 @@ class PerfCounters final {
   // names()[i]'s value is (*values)[i]
   BENCHMARK_ALWAYS_INLINE bool Snapshot(PerfCounterValues* values) {
 #ifndef BENCHMARK_OS_WINDOWS
-    assert(values != nullptr);
-    assert(IsValid());
-    auto buffer = values->get_data_buffer();
-    auto read_bytes = ::read(counter_ids_[0], buffer.first, buffer.second);
-    return static_cast<size_t>(read_bytes) == buffer.second;
+    if (values != nullptr && IsValid()) {
```

Comment:
why is this better than asserting?

Reply:
If BENCHMARK_OS_WINDOWS is defined, then it simply returns false without any asserts. This seems to me as non-uniform behavior.

But this change is not necessary and can be reverted.

Reply:
This shouldn't even be called on windows or on cases where the counters aren't supported; the windows test is to circumvent dealing with ::read's definition being different on windows.

That aside, I agree with you, why not have the behavior consistent - I'd propose having this in a different patch, though, it's not logging - related.